### ISSUE TITLE
msfdb init/reinit fails,(bundler: failed to load command: rake)

### ISSUE THREAD

**Issue by raypun78** [rapid7/metasploit-framework#15626]
How'd you do it?

1. type msfdb init and create a new config file, since i deleted it. also start postgresql service.

This section should also tell us any relevant information about the
environment; for example, if an exploit that used to work is failing,
tell us the victim operating system and service versions.

## Were you following a specific guide/tutorial or reading documentation?

https://www.youtube.com/watch?v=r3DPN8XLEnI

also i read lots of other articles about msfdb reinit and gems because armitage was giving serious error about database.yml file even tho i had it in the proper directory.

## Expected behavior

should create a proper config without errors

## Current behavior

i firstly have following error: bundler: failed to load command: rake (/opt/metasploit-framework/embedded/bin/rake)

then 

Could not find jekyll-4.2.0, colorator-1.1.0, em-websocket-0.5.2, jekyll-sass-converter-2.1.0, jekyll-watch-2.2.1, kramdown-2.3.1, kramdown-parser-gfm-1.1.0, liquid-4.0.3, mercenary-0.4.0, pathutil-0.16.2, rouge-3.26.0, safe_yaml-1.0.5, terminal-table-2.0.0, http_parser.rb-0.6.0, concurrent-ruby-1.1.9, sassc-2.4.0, listen-3.7.0, forwardable-extended-2.6.0, unicode-display_width-1.7.0, rb-fsevent-0.11.0, rb-inotify-0.10.1 in any of the sources (Bundler::GemNotFound)

### Metasploit version

Framework: 6.1.3-dev-
Console  : 6.1.3-dev-

### **TERMINAL INPUT AND OUTPUT**

```
(root💀kali)-[~]# msfdb init

Metasploit running on Kali Linux as root, using system database
Resetting password of database user 'msf'
Creating databases 'msf' and 'msf_test'
Creating configuration file in /usr/share/metasploit-framework/config/database.yml
Creating initial database schema

bundler: failed to load command: rake (/opt/metasploit-framework/embedded/bin/rake)
Traceback (most recent call last):
        22: from /opt/metasploit-framework/embedded/bin/bundle:23:in `<main>'
        21: from /opt/metasploit-framework/embedded/bin/bundle:23:in `load'
        20: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/exe/bundle:37:in `<top (required)>'
        19: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/friendly_errors.rb:128:in `with_friendly_errors'
        18: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/exe/bundle:49:in `block in <top (required)>'
        17: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli.rb:25:in `start'
        16: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        15: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli.rb:31:in `dispatch'
        14: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        13: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        12: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        11: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli.rb:477:in `exec'
        10: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli/exec.rb:23:in `run'
         9: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli/exec.rb:56:in `kernel_load'
         8: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli/exec.rb:56:in `require_relative'
         7: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/setup.rb:20:in `<top (required)>'
         6: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/ui/shell.rb:88:in `silence'
         5: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/ui/shell.rb:136:in `with_level'
         4: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/setup.rb:20:in `block in <top (required)>'
         3: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler.rb:149:in `setup'
         2: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/runtime.rb:18:in `setup'
         1: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/definition.rb:228:in `specs_for'
         
/opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/definition.rb:490:in `materialize': Could not find jekyll-4.2.0, colorator-1.1.0, em-websocket-0.5.2, jekyll-sass-converter-2.1.0, jekyll-watch-2.2.1, kramdown-2.3.1, kramdown-parser-gfm-1.1.0, liquid-4.0.3, mercenary-0.4.0, pathutil-0.16.2, rouge-3.26.0, safe_yaml-1.0.5, terminal-table-2.0.0, http_parser.rb-0.6.0, concurrent-ruby-1.1.9, sassc-2.4.0, listen-3.7.0, forwardable-extended-2.6.0, unicode-display_width-1.7.0, rb-fsevent-0.11.0, rb-inotify-0.10.1 in any of the sources (Bundler::GemNotFound)`
```

### I installed gems and typed required commands like gem install, gem upgrade, bundler install and installed rake manually, and when i typed rake it says Rakefile can not be found but i couldnt have any solutions.

**Comment by adfoster-r7** [comment]
It looks like your gems are missing, this fix may work for you: https://github.com/rapid7/metasploit-framework/issues/11597#issuecomment-474940871

**Comment by raypun78** [comment]
> It looks like your gems are missing, this fix may work for you: [#11597 (comment)](https://github.com/rapid7/metasploit-framework/issues/11597#issuecomment-474940871)

hello, i already read that issue's comments and done nothing with it. when i use "gem update --system" it gives the following error: 

`(root💀kali)-[/usr/share/metasploit-framework]# gem update --system
ERROR:  Your RubyGems was installed trough APT, and upgrading it through RubyGems
itself is unsupported. If you really need the latest version of RubyGems (tip:
you usually don't), then you need to install RubyGems (and Ruby) manually,
maybe using tools like ruby-install, rvm, etc.`

I looked about how to download ruby manually but it was helpless. I tryed using rvm but even that i got the rvm keys, rvm command was undefined in my machine. I need a little further help

**Comment by raypun78** [comment]
> It looks like your gems are missing, this fix may work for you: [#11597 (comment)](https://github.com/rapid7/metasploit-framework/issues/11597#issuecomment-474940871)

i reinstalled ruby with rvm then ran first 2 commands but at the last command, i got the following error:

`root💀kali)-[/usr/share/metasploit-framework]# gem update --system                                            1 ⨯
Error loading RubyGems plugin "/usr/local/rvm/gems/ruby-3.0.0/plugins/executable-hooks_plugin.rb": cannot load such file -- /usr/local/rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/executable-hooks-1.6.1/lib/rubygems_plugin.rb (LoadError)`
then it updated something and when i retyped it it said everything is updated. Lastly i ran msfdb reinit to remake the database.yml file but i still get the same error:

`(root💀kali)-[/usr/share/metasploit-framework]# msfdb reinit
Metasploit running on Kali Linux as root, using system database
┏━(Message from Kali developers)
┃
┃ This is a minimal installation of Kali Linux, you likely
┃ want to install supplementary tools. Learn how:
┃ ⇒ https://www.kali.org/docs/troubleshooting/common-minimum-setup/
┃
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
┃
┗━(Run: “touch ~/.hushlogin” to hide this message)
┏━(Message from Kali developers)
┃
┃ This is a minimal installation of Kali Linux, you likely
┃ want to install supplementary tools. Learn how:
┃ ⇒ https://www.kali.org/docs/troubleshooting/common-minimum-setup/
┃
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
┃
┗━(Run: “touch ~/.hushlogin” to hide this message)
┏━(Message from Kali developers)
┃
┃ This is a minimal installation of Kali Linux, you likely
┃ want to install supplementary tools. Learn how:
┃ ⇒ https://www.kali.org/docs/troubleshooting/common-minimum-setup/
┃
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
┃
┗━(Run: “touch ~/.hushlogin” to hide this message)
Creating database user 'msf'
Enter password for new role: 
Enter it again: 
Creating databases 'msf' and 'msf_test'
┏━(Message from Kali developers)
┃
┃ This is a minimal installation of Kali Linux, you likely
┃ want to install supplementary tools. Learn how:
┃ ⇒ https://www.kali.org/docs/troubleshooting/common-minimum-setup/
┃
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
┃
┗━(Run: “touch ~/.hushlogin” to hide this message)
┏━(Message from Kali developers)
┃
┃ This is a minimal installation of Kali Linux, you likely
┃ want to install supplementary tools. Learn how:
┃ ⇒ https://www.kali.org/docs/troubleshooting/common-minimum-setup/
┃
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
┃
┗━(Run: “touch ~/.hushlogin” to hide this message)
Creating configuration file in /usr/share/metasploit-framework/config/database.yml
Creating initial database schema
bundler: failed to load command: rake (/opt/metasploit-framework/embedded/bin/rake)
Traceback (most recent call last):
        22: from /opt/metasploit-framework/embedded/bin/bundle:23:in `<main>'
        21: from /opt/metasploit-framework/embedded/bin/bundle:23:in `load'
        20: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/exe/bundle:37:in `<top (required)>'
        19: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/friendly_errors.rb:128:in `with_friendly_errors'
        18: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/exe/bundle:49:in `block in <top (required)>'
        17: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli.rb:25:in `start'
        16: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        15: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli.rb:31:in `dispatch'
        14: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        13: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        12: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        11: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli.rb:477:in `exec'
        10: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli/exec.rb:23:in `run'
         9: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli/exec.rb:56:in `kernel_load'
         8: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/cli/exec.rb:56:in `require_relative'
         7: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/setup.rb:20:in `<top (required)>'
         6: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/ui/shell.rb:88:in `silence'
         5: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/ui/shell.rb:136:in `with_level'
         4: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/setup.rb:20:in `block in <top (required)>'
         3: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler.rb:149:in `setup'
         2: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/runtime.rb:18:in `setup'
         1: from /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/definition.rb:228:in `specs_for'
/opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/bundler-2.2.26/lib/bundler/definition.rb:490:in `materialize': Could not find jekyll-4.2.0, colorator-1.1.0, em-websocket-0.5.2, jekyll-sass-converter-2.1.0, jekyll-watch-2.2.1, kramdown-2.3.1, kramdown-parser-gfm-1.1.0, liquid-4.0.3, mercenary-0.4.0, pathutil-0.16.2, rouge-3.26.0, safe_yaml-1.0.5, terminal-table-2.0.0, http_parser.rb-0.6.0, concurrent-ruby-1.1.9, sassc-2.4.0, listen-3.7.0, forwardable-extended-2.6.0, unicode-display_width-1.7.0, rb-fsevent-0.11.0, rb-inotify-0.10.1 in any of the sources (Bundler::GemNotFound)`

by the way i did these in /usr/share/metasploit-framework/ directory.

**Comment by raypun78** [comment]
> It looks like your gems are missing, this fix may work for you: [#11597 (comment)](https://github.com/rapid7/metasploit-framework/issues/11597#issuecomment-474940871)

hello again, i removed ruby fully and reinstalled with rvm. now i get no error while using "gem update --system" but i still get same error with "msfdb init" or "msfdb reinit"
